rest/client.go

```diff
@@ -119,7 +120,7 @@ var do = func(c *http.Client, method, url string, body interface{}, header http.
 func decodeResponse(response []byte, r *http.Response) (*Response, error) {
 	var payload map[string]interface{}
 	if err := json.Unmarshal(response, &payload); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%v. Response with status %d: %s had unmarshallable payload %s", err, r.StatusCode, r.Status, string(response))
```

Comment:
You might instead create a custom error type that allows extraction of the inner error instead of just flattening into a String.  The `String()` of the new error type would generate this string of course.